lib/liquid/block_body.rb

```diff
@@ -67,19 +67,23 @@ def blank?
     end
 
     def render(context)
-      output = []
+      render_to_output_buffer(context, '')
+    end
+
+    def render_to_output_buffer(context, output)
       context.resource_limits.render_score += @nodelist.length
 
       idx = 0
       while node = @nodelist[idx]
+        previous_output_size = output.bytesize
+
         case node
         when String
-          check_resources(context, node)
           output << node
         when Variable
-          render_node_to_output(node, output, context)
+          render_node(context, output, node)
         when Block
-          render_node_to_output(node, output, context, node.blank?)
+          render_node(context, node.blank? ? '' : output, node)
```

Comment:
I've always found blank handling a bit confusing, what do you think about making the block responsible for not writing anything if it's blank, instead of writing to a temporary buffer that we'll just throw away? Can ignore for this PR in any case.

Reply:
Yea agree, have it on my TODO list to look into that as a follow-up because I saw this `''` allocation come up in the memory profile quite a lot. Will ignore for this PR.